lib/bashcov/bash_info.rb

```diff
@@ -0,0 +1,30 @@
+module Bashcov
+  # Module exposing information concerning the installed Bash version
+  # @note methods do not cache results because +bash_path+ can change at
+  #   runtime
+  # @note receiver is expected to implement +bash_path+
+  module BashInfo
+    # @return [Array<String>] An array representing the components of
+    #   +BASH_VERSINFO+
+    def bash_versinfo
+      `#{bash_path} -c 'echo "${BASH_VERSINFO[@]}"'`.chomp.split
+    end
+
+    # @return [Boolean] Whether Bash supports +BASH_XTRACEFD+
+    def bash_xtracefd?
+      bash_versinfo[0..1].join.to_i >= 41
```

Comment:
I'm tempted to NOT support very old versions of Bash. [4.1 was shipped up to Debian Squeeze](https://packages.debian.org/search?keywords=bash&searchon=names&exact=1&suite=all&section=all) which was removed from security support this month.

In this case, it doesn't add many `if` branches nor much overhead complexity to maintain so I think it's fine to keep it for now. However I don't want Bashcov codebase to be twice as big just to support obsolete versions of Bash ;)